src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/remote/api/SimpleItemApi.kt

```diff
@@ -0,0 +1,181 @@
+import net.minecraft.world.item.Items
+import tech.thatgravyboat.repolib.api.RepoAPI
+import tech.thatgravyboat.skyblockapi.api.SkyBlockAPI
+import tech.thatgravyboat.skyblockapi.api.data.SkyBlockRarity
+import tech.thatgravyboat.skyblockapi.api.remote.PetQuery
+import tech.thatgravyboat.skyblockapi.api.remote.RepoItemsAPI
+import tech.thatgravyboat.skyblockapi.api.remote.RepoPetsAPI
+import tech.thatgravyboat.skyblockapi.api.remote.RepoRunesAPI
+import tech.thatgravyboat.skyblockapi.api.remote.api.SkyBlockId.Companion.UNKNOWN
+import tech.thatgravyboat.skyblockapi.api.remote.api.SkyBlockId.Companion.attribute
+import tech.thatgravyboat.skyblockapi.api.remote.api.SkyBlockId.Companion.enchantment
+import tech.thatgravyboat.skyblockapi.api.remote.api.SkyBlockId.Companion.item
+import tech.thatgravyboat.skyblockapi.api.remote.api.SkyBlockId.Companion.pet
+import tech.thatgravyboat.skyblockapi.api.remote.api.SkyBlockId.Companion.rune
+import tech.thatgravyboat.skyblockapi.utils.builders.ItemBuilder
+import tech.thatgravyboat.skyblockapi.utils.extentions.sanitizeForCommandInput
+import tech.thatgravyboat.skyblockapi.utils.extentions.stripColor
+import tech.thatgravyboat.skyblockapi.utils.extentions.toIntValue
+import tech.thatgravyboat.skyblockapi.utils.extentions.toReadableTime
+import tech.thatgravyboat.skyblockapi.utils.json.getPath
+import tech.thatgravyboat.skyblockapi.utils.time.currentInstant
+import tech.thatgravyboat.skyblockapi.utils.time.since
+
+object SimpleItemApi {
+
+    private val cache: MutableMap<SkyBlockId, ItemStack?> = mutableMapOf()
+    private val nameCache: MutableMap<String, SkyBlockId> = mutableMapOf()
+
+    init {
+        val start = currentInstant()
+        RepoAPI.pets().pets().entries.associate { (id, data) -> data.name() to pet(id) }
+            .let(nameCache::putAll)
+
+        RepoAPI.runes().runes().entries.flatMap { (id, data) ->
+            data.map { rune ->
+                rune.name().stripColor() to rune("$id:${rune.tier()}")
+            }
+        }.toMap().let(nameCache::putAll)
+
+        RepoAPI.enchantments().enchantments().flatMap { (id, enchantments) ->
+            enchantments.levels().map { (level, enchantment) ->
+                "${enchantments.name()} ${enchantment.literalLevel()}" to enchantment("$id:${enchantment.level()}")
+            }
+        }.toMap().let(nameCache::putAll)
+
+        RepoAPI.attributes().attributes().flatMap { (id, attribute) ->
+            listOf(
+                attribute.name() to attribute(attribute.id()),
+                attribute.shardName() to attribute(attribute.id()),
+            )
+        }.toMap().let(nameCache::putAll)
+
+        RepoAPI.items().items().entries.mapNotNull { (id, element) ->
+            val components =
+                element.getPath("['components'].['minecraft:custom_name'].['text']") ?: return@mapNotNull null
+            components.asString.stripColor() to item(id)
+        }.toMap().let(nameCache::putAll)
```

Comment:
instead of transforming to a map can you do the nameCache.put inside and use forEach instead

Reply:
That does make it like 1249127912x slower